### ISSUE TITLE
Bot's not merging

### ISSUE THREAD

**Issue by zregvart** [redhat-ipaas/pure-bot#63]
I've noticed on a couple of PRs, seems that the bot is not merging approved PRs any more. One example is syndesisio/syndesis#4213. Could be related to the CircleCI [checks](https://github.com/syndesisio/syndesis/pull/4213/checks) thing change I made.

I've found no errors the log.
[pure-bot-55-56xqz.log](https://github.com/syndesisio/pure-bot/files/2666971/pure-bot-55-56xqz.log)

**Comment by zregvart** [comment]
After a bit of `curl`-ing:

```sh
$ curl https://api.github.com/repos/syndesisio/syndesis/commits/a0a56f2716610a7ee97734b40c5665d3bdc7f2bf/status
{
  "state": "success",
  "statuses": [
    {
      "url": "https://api.github.com/repos/syndesisio/syndesis/statuses/a0a56f2716610a7ee97734b40c5665d3bdc7f2bf",
      "avatar_url": "https://avatars3.githubusercontent.com/u/23079786?v=4",
      "id": 5930700555,
      "node_id": "MDEzOlN0YXR1c0NvbnRleHQ1OTMwNzAwNTU1",
      "state": "success",
      "description": "OK - this is not a work in progress",
      "target_url": null,
      "context": "pure-bot/wip",
      "created_at": "2018-12-10T20:31:47Z",
      "updated_at": "2018-12-10T20:31:47Z"
    },
    {
      "url": "https://api.github.com/repos/syndesisio/syndesis/statuses/a0a56f2716610a7ee97734b40c5665d3bdc7f2bf",
      "avatar_url": "https://avatars3.githubusercontent.com/u/23079786?v=4",
      "id": 5933891675,
      "node_id": "MDEzOlN0YXR1c0NvbnRleHQ1OTMzODkxNjc1",
      "state": "success",
      "description": "OK - review requested and at least one provided",
      "target_url": null,
      "context": "pure-bot/pr-review",
      "created_at": "2018-12-11T09:40:12Z",
      "updated_at": "2018-12-11T09:40:12Z"
    }
  ],
  "sha": "a0a56f2716610a7ee97734b40c5665d3bdc7f2bf",
  "total_count": 2,
...
}

$ curl -u zregvart:... -H 'Accept: application/vnd.github.luke-cage-preview' https://api.github.com/repos/syndesisio/syndesis/branches/master/protection/required_status_checks
{
  "url": "https://api.github.com/repos/syndesisio/syndesis/branches/master/protection/required_status_checks",
  "strict": false,
  "contexts": [
    "pure-bot/pr-review",
    "pure-bot/wip",
    "syndesis"
  ],
  "contexts_url": "https://api.github.com/repos/syndesisio/syndesis/branches/master/protection/required_status_checks/contexts"
}
```

That means that this has two statuses:

https://github.com/syndesisio/pure-bot/blob/f2aa11a38f16f7fea68521950358117bb52c0ad7/pkg/webhook/auto_merge.go#L128

And here we require three:

https://github.com/syndesisio/pure-bot/blob/f2aa11a38f16f7fea68521950358117bb52c0ad7/pkg/webhook/auto_merge.go#L138

The check status for `syndesis` required status check is not present in the first response but is required for the merge to proceed.

**Comment by rhuss** [comment]
Fixed with #64
